apps/e2e/tests/utils/mailbox.ts

```
@@ -41,47 +35,67 @@ export class Mailbox {
+      const json = (await response.json()) as { messages: Array<{ ID: string }> };
 
-    const messageId = json[0]?.id;
-    const messageUrl = `${url}/${messageId}`;
+      if (json?.messages && json.messages.length > 0) {
+        console.log(`Email found after ${attempts + 1} attempts`);
+        
+        // Get the most recent email (first in the list)
+        const messageId = json.messages[0]?.ID;
+        if (!messageId) {  
+          throw new Error('Email message ID not found');  
++       }  
```

Biome (1.9.4)

[error] This code is unreachable

... because this statement will throw an exception beforehand


(lint/correctness/noUnreachable)